### DISCUSSION TITLE
Tap-hold behavior

### DISCUSSION THREAD

**Discussion by yarnaid** [pqrs-org/Karabiner-Elements#4026] [Q&A] [answered]
I want to know if there is a way I can get different behavior for a short and long press of a key. E.g. I want to have on hold F it becomes shift. And on tap it's still F. It's a home row mod from Miroyko. Unfortunately, it doesn't work properly with if_held_down if I type quickly. Any ideas how to fix this problem?

**Comment by tekezo** [comment]
While it's not a "hold", using `to_if_alone` with a shorter `basic.to_if_alone_timeout_milliseconds` could be a good fit for your intended use.
https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/to-if-alone/#about-cancellation

Alternatively, simple_vi_mode implements this using simultaneous key presses.
This makes s+hjkl act as arrow keys. Of course, if you press s alone, it will remain as the s key.

https://github.com/pqrs-org/KE-complex_modifications/blob/main/public/json/personal_tekezo_simple_vi_mode.json
https://github.com/pqrs-org/KE-complex_modifications/blob/main/src/json/personal_tekezo_simple_vi_mode.json.js

**Reply by yarnaid** [reply]
Unfortunately, this doesn't work for me. If I press, e.g., fj very quickly, I want them to be printed even if F was down (less than 200) simultaneously with J.

**Reply by tekezo** [reply, marked as answer]
Thank you for your response. If triggering on a long press is absolutely required, a more complex configuration will be necessary.
I’ve added an example, "Change the f key to the left shift key if held down" Please take a look.

https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/to-if-held-down/#more-advanced-example

**Reply by yarnaid** [reply]
Wow! Works like a charm!
